S100-Permit-Service-Open-Api-Spec-v0.1.yaml

```
@@ -2,38 +2,101 @@ openapi: 3.0.1
 info:
   title: UKHO S100 Permit Service API
   description: |
-    The UKHO S100 Permit Service API.
+    This API provides functionality to generate signed PERMIT.XML files 
+    using the S-100 Part 15 data protection scheme. It enables users to request permits 
+    for specific products and retrieve them as a compressed ZIP file.
   contact:
     name: UKHO
     email: abzudeliveryteam@ukho.gov.uk
   version: 0.0.1
   
 externalDocs:
   url: https://github.com/UKHO/s-100-permit-service
+  
 servers:
-  - url: https://permit-service-url
-    description: Proposed URL
+  - url: https://{environment}.s100.permit-service.com
+    description: UKHO Permit Service API Environment
+    variables:
+      environment:
+        default: local
+        enum:
+          - local
+          - dev
+          - vnextiat
+          - vnexte2e
+          - iat
+          - preProd
+          - live
 
 security:
   - jwtBearerAuth: []
   
 paths:
-  /permits/{licenceId}:
-    get:
+  /v1/permits/s100:
+    post:
       description: |
-        It uses the S-100 Part 15 data protection scheme to generate signed PERMIT.XML files for all the User Permit Numbers (UPNs) for the requested licence and returns a compressed zip file containing all these PERMIT.XML files.
+        Generates signed PERMIT.XML files for requested products and user permits (UPNs),
+        returning them in a compressed ZIP file.
       operationId: getS100Permits
       parameters:
-      - name: licenceId
-        in: path
-        description: Licence Id. It must be an integer value and greater than zero.
+        - name: Authorisation
+          in: header
+          required: true
+          schema:
+            type: string
+          description: "Bearer token, where token is a valid JWT token"
+        - name: X-Correlation-ID
+          in: header
+          required: false
+          schema:
+            type: string
+          description: Request tracing identifier      
+      requestBody:
+        description: |
+          The JSON body containing products and user permits, for which the permits are to be generated.
         required: true
-        schema:
-          type: integer
-          example: 12345678
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                products:
+                  type: array
+                  items:
+                    $ref: '#/components/schemas/product'
+                userPermits:
+                  type: array
+                  items:
+                    $ref: '#/components/schemas/userPermit'
```

Suggest adding an indication that the array can't be empty

```suggestion
                    $ref: '#/components/schemas/userPermit'
                  minItems: 1
```